platform/linux-generic/odp_packet.c

```diff
@@ -2506,6 +2506,16 @@ int odp_packet_has_ref(odp_packet_t pkt)
 	return 0;
 }
 
+odp_proto_l2_type_t odp_packet_l2_type(odp_packet_t pkt)
+{
+	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
+
+	if (pkt_hdr->p.input_flags.eth)
+		return ODP_PROTO_L2_TYPE_ETH;
+
+	return ODP_PROTO_L2_TYPE_NONE;
```

Comment:
`return pkt_hdr->p.input_flags.eth ? ODP_PROTO_L2_TYPE_ETH : ODP_PROTO_L2_TYPE_NONE;` might be simpler here.

Reply:
No, because this code is implemented in the same way as the rest of `odp_packet_lX_type()` functions.